src/plugins/query_enhancements/public/query_assist/utils/create_extension.tsx

```diff
@@ -6,29 +6,30 @@
 import { HttpSetup } from 'opensearch-dashboards/public';
 import React, { useEffect, useState } from 'react';
 import { of } from 'rxjs';
-import { distinctUntilChanged, switchMap, map } from 'rxjs/operators';
+import { distinctUntilChanged, map, startWith, switchMap } from 'rxjs/operators';
 import {
+  DataPublicPluginSetup,
   QueryEditorExtensionConfig,
   QueryEditorExtensionDependencies,
 } from '../../../../data/public';
 import { API } from '../../../common';
 import { ConfigSchema } from '../../../common/config';
-import { ConnectionsService } from '../../services';
-import { QueryAssistBar, QueryAssistBanner } from '../components';
+import { QueryAssistBanner, QueryAssistBar } from '../components';
 
 /**
  * @returns observable list of query assist agent configured languages in the
  * selected data source.
  */
 const getAvailableLanguages$ = (
   availableLanguagesByDataSource: Map<string | undefined, string[]>,
-  connectionsService: ConnectionsService,
-  http: HttpSetup
+  http: HttpSetup,
+  data: DataPublicPluginSetup
 ) =>
-  connectionsService.getSelectedConnection$().pipe(
+  data.query.dataSet.getUpdates$().pipe(
+    startWith(data.query.dataSet.getDataSet()),
```

Comment:
@kavilla why is there a skip 1 in dataset manager? i had to add a startWith to make it work
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/adbdd9501846cb4062697c4dabc79330eb1ddd9e/src/plugins/data/public/query/dataset_manager/dataset_manager.ts#L29

Reply:
He copied it over from the query manager. Yeah i dont think it needs to be there either. Feel free to remove it. We need the value of the dataset immediately

Reply:
+1

Reply:
thanks for confirming, i removed it